### PR TITLE
add darkheart thicket trash module

### PR DESCRIPTION
**Pull request by roionsteroids**
very tame, not spammy ;)

### CODE REVIEW

File: Legion/TheArcway/Trash.lua [hunk 1]
```diff
@@ -28,29 +32,42 @@ L = mod:GetLocale()
 
 function mod:GetOptions()
 	return {
-		--[[ Arcane Anomaly ]]--
+		-- Arcane Anomaly
```
Comment (elvador) [inline Legion/TheArcway/Trash.lua]: Why are you removing those brackets? We are using them as headers.

File: Legion/DarkheartThicket/Trash.lua [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+	self:RegisterMessage("BigWigs_OnBossEngage", "Disable")
+
+	-- Dreadsoul Ruiner
+	self:Log("SPELL_CAST_START", "StarShower", 200658)
+
+	-- Crazed Razorbeak
+	self:Log("SPELL_CAST_START", "PropellingCharge", 200768)
+
+	-- Festerhide Grizzly
+	self:Log("SPELL_CAST_START", "MaddeningRoar", 200580)
```
Comment (elvador) [inline Legion/DarkheartThicket/Trash.lua]: `MaddeningRoar` function does not exist.

**Comment by roionsteroids** [comment]
Every module written by a different person uses some kind of different style, that specific one (brackets) doesn't really exist in any other trash module I've looked at?

Added the MaddeningRoar function as requested.

**Comment by elvador** [comment]
Just because it's not strictly enforced doesn't mean they should be removed. Just follow the markup the module you're editing uses.

Here are a few examples of trash modules using this style:
https://github.com/BigWigsMods/LittleWigs/blob/master/Legion/EyeOfAzshara/Trash.lua
https://github.com/BigWigsMods/LittleWigs/blob/master/Legion/NeltharionsLair/Trash.lua
https://github.com/BigWigsMods/LittleWigs/blob/master/Legion/TheArcway/Trash.lua
https://github.com/BigWigsMods/LittleWigs/blob/master/Legion/VaultOfTheWardens/Trash.lua

**Comment by funkydude** [comment]
That's a lot of of functions using the exact same colours and sounds. If these are all separate mobs that aren't pulled together, that's fine, but merge the functions.

If a lot of these warnings will be overlapping, you'll need to vary it.

**Comment by roionsteroids** [comment]
> Just because it's not strictly enforced doesn't mean they should be removed.

Not sure how exactly the brackets are helpful, generally I only see them being used for multi-line comments (which isn't the case here?). But since you appear to be seriously triggered by the lack of brackets, I've added them back.

> If these are all separate mobs that aren't pulled together, that's fine, but merge the functions.

Done.

File: Legion/TheArcway/Trash.lua [hunk 3]
```diff
@@ -28,46 +32,71 @@ L = mod:GetLocale()
 
 function mod:GetOptions()
 	return {
-		--[[ Arcane Anomaly ]]--
+		-- [[Arcane Anomaly]]
```
Comment (elvador) [inline Legion/TheArcway/Trash.lua]: ...

**Comment by funkydude** [comment]
``--[[ This is a header ]]--``
``--This is a basic comment``

There is no functional difference, the difference is it becomes a lot easier to separate actual comments from categorisation.

Please remove the arcway changes and post them as a separate PR, thanks.

**Comment by roionsteroids** [comment]
As requested, I restored the formatting in the arcway module and implemented the same format for the new darkheart trash module.

I don't think it's very helpful to lose our heads over semantics though, and instead focus on the actual additions, any concerns about that at all? The spell IDs are all correct, there are no errors in the ingame config and everything appeared to be working as expected in test runs. Kept it very basic and only included the (very few) most troublesome abilities.

**Comment by funkydude** [comment]
I have no objections against the actual content of the module. However I'd like to see the arcway changes split into a separate PR.

As a general rule of thumb to keep things as fast and smooth and possible, just copy paste other modules with the formatting intact and make the functional changes needed. It reduces work for us all and speeds up merging.

**Comment by roionsteroids** [comment]
I've no idea how to do that actually (separate commits into different PRs), and hmm well does it make any difference if you're accepting both anyway?

**Comment by funkydude** [comment]
You can create multiple PRs by using multiple branches. Just create a new branch on your repo, copy paste the changes, submit a PR.

Copy the default module into the current branch and commit it, it should basically eliminate the changes in the diff i think?

**Pull request merged**